### PR TITLE
Void 3.0 subclass fixes

### PR DESCRIPTION
**Pull request by robojumper**
Void 3.0 crashes beta LO upon selection, Echo of Persistence gets the stats wrong, and there are some other easy fixes.

### CODE REVIEW

File: src/app/loadout-builder/loadout-builder-reducer.ts [hunk 1]
```diff
@@ -114,21 +114,13 @@ const lbStateInit = ({
...
-            const socketOverridesForLO = { ...loadoutItem.socketOverrides };
-
             // In LO we populate the default ability plugs because in game you cannot unselect all abilities.
-            for (const socket of abilityAndSuperSockets) {
-              if (!socketOverridesForLO[socket.socketIndex]) {
-                socketOverridesForLO[socket.socketIndex] =
-                  socket.socketDefinition.singleInitialItemHash;
-              }
-            }
-            subclass = { ...item, socketOverrides: loadoutItem.socketOverrides };
```
Comment (robojumper) [inline src/app/loadout-builder/loadout-builder-reducer.ts:131]: The unused `socketOverridesForLO` seemed like a bug?

Comment (ryan-rushton) [reply src/app/loadout-builder/loadout-builder-reducer.ts:131]: Yeah, interesting that it didn't cause issues. From what I can see it should mean default socket overrides don't get set. We are probably checking that in a few areas which means that it doesn't matter it gets missed here.

File: src/app/loadout-drawer/loadout-utils.ts [hunk 2]
```diff
@@ -90,6 +91,27 @@ export function createSocketOverridesFromEquipped(item: DimItem) {
...
+  if (item.bucket.hash === BucketHashes.Subclass && item.sockets) {
+    const socketOverrides: SocketOverrides = {};
+    const abilityAndSuperSockets = getSocketsByCategoryHashes(item.sockets, [
+      SocketCategoryHashes.Abilities_Abilities_DarkSubclass,
+      SocketCategoryHashes.Abilities_Abilities_LightSubclass,
+      SocketCategoryHashes.Super,
+    ]);
+
+    for (const socket of abilityAndSuperSockets) {
+      // HACK: Void grenades do not have a singleInitialItemHash
```
Comment (ryan-rushton) [inline src/app/loadout-drawer/loadout-utils.ts:107]: Jeez there are a few holes in the dataset for subclasses

File: src/app/loadout-drawer/loadout-utils.ts [hunk 3]
```diff
@@ -213,7 +235,10 @@ export function getLoadoutStats(
     for (const plugHash of Object.values(subclass.socketOverrides)) {
       const plug = defs.InventoryItem.get(plugHash);
       for (const stat of plug.investmentStats) {
-        if (stat.statTypeHash in stats) {
+        if (
+          stat.statTypeHash in stats &&
+          isPlugStatActive(subclass, plugHash, stat.statTypeHash, stat.isConditionallyActive)
+        ) {
```
Comment (ryan-rushton) [inline src/app/loadout-drawer/loadout-utils.ts:241]: Are any subclass fragments conditionally active?

**Review by ryan-rushton — APPROVED** [review]
Looks good and I can't break it in any way.

**Comment by ryan-rushton** [comment]
Merging this since it fixes beta. Did this exist before #7945? If so we might want to add a change log.

**Pull request merged**